ql/src/test/org/apache/hadoop/hive/ql/io/parquet/serde/TestParquetTimestampsHive2Compatibility.java

```diff
@@ -79,6 +79,18 @@ void testWriteHive2ReadHive4UsingLegacyConversion(String timestampString) {
     assertEquals(timestampString, ts.toString());
   }
 
+  /**
+   * Tests that timestamps written using Hive2 APIs are read correctly by Hive4 APIs when legacy conversion is on.
+   */
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("generateTimestamps")
+  void testWriteHive2ReadHive4UsingLegacyConversionWithZone(String timestampString) {
+    String zoneId = "US/Pacific";
+    NanoTime nt = writeHive2(timestampString);
```

Comment:
Since there is no parameter in `writeHive2` for specifying the timezone I think you will need to call `TimeZone.setDefault()` explicitly otherwise it will not work.